integrations/docker/images/chip-build-esp32/Dockerfile

```diff
@@ -7,7 +7,7 @@ RUN set -x \
     && DEBIAN_FRONTEND=noninteractive apt-get install -y python \
     && mkdir -p /opt/espressif \
     && cd /opt/espressif \
-    && git clone --progress -b release/v4.1 https://github.com/espressif/esp-idf.git \
+    && git clone --progress -b release/v4.2 https://github.com/espressif/esp-idf.git \
```

Comment:
you need to update dockerfile version for this.

Reply:
And then a followup PR will be required to update all our CI and vscode to the new version.

Logic being that we want to test compilation and docker CI images will also be used to generate release artifacts.

Reply:
@andy31415  I have incremented the dockerfile version